js/chatbot/shared/ChatBot.svelte

```diff
@@ -482,13 +483,15 @@
 						{/each}
 					</div>
 				</div>
+				{@const show_like =
+					role === "user" ? likeable && like_user_message : likeable}
 				<LikeButtons
-					show={likeable ||
+					show={show_like ||
 						(_retryable && is_last_bot_message(messages, value)) ||
 						(_undoable && is_last_bot_message(messages, value)) ||
 						show_copy_button}
 					handle_action={(selected) => handle_like(i, messages[0], selected)}
-					{likeable}
+					likeable={show_like}
 					_retryable={_retryable && is_last_bot_message(messages, value)}
 					_undoable={_undoable && is_last_bot_message(messages, value)}
```

Comment:
Just thought that `_retryable` and `_undoable` can also be defined as variables like `@const show_like` to remove the duplications of `_retryable && is_last_bot_message(messages, value)` and `_undoable && is_last_bot_message(messages, value)` respectively as refactoring, though it's not related to this PR.

Reply:
Yes great idea!